app/Models/Contact/Contact.php

```diff
@@ -162,6 +163,26 @@ public function account()
         return $this->belongsTo(Account::class);
     }
 
+    /**
+     * Get the address book associated with the contact.
+     *
+     * @return BelongsTo
+     */
+    public function addressBook()
+    {
+        return $this->belongsTo(AddressBook::class);
+    }
+
+    /**
+     * Get the list of contacts from the same address book as this contact.
+     *
+     * @return HasMany|null
+     */
+    public function addressBookContacts()
+    {
+        return $this->account ? $this->account->addressBookContacts($this->addressBook ? $this->addressBook->name : null) : null;
```

Comment:
Way too complex in my opinion.
You are trying to be too smart 😀 Yes the code is more pure that way, but it comes with a price of having to decrypt the line to understand what it does.

Could you make it multi line so it’s easy to read in one glance?